httpserver/server.go

```diff
@@ -87,6 +88,7 @@ func NewServer(cfg Config) (*Server, error) {
 
 	s := &Server{
 		schedd:         schedd,
+		collector:      cfg.Collector,
```

Comment:
In the `cmd/htcondor-api/main.go`, populate the structure as well, falling back to the global config if not provided in a command-line flag.